src/pharmpy/modeling/data.py

```diff
@@ -964,38 +964,43 @@ def set_admid(model: Model, column_name: str):
     return model.update_source()
 
 
-def get_cmt(model: Model):
+def get_cmt(model: Model, force: bool = False):
     """Get the cmt (compartment) column from the model dataset
 
     If a cmt column is present this will be extracted otherwise
     a cmt column will be created. If created, multiple dose compartments are
     dependent on the presence of an admid type column, otherwise, dose/non-dose
     will be considered.
 
+    If force, a new CMT column will forcefully be created
+
     Parameters
     ----------
     model : Model
         Pharmpy model
+    force : bool
+        Forcefully create a new CMT column
 
     Returns
     -------
     pd.Series
         CMT
     """
     di = model.datainfo
-    try:
-        cmtcols = di.typeix['compartment']
-    except IndexError:
-        pass
-    else:
-        return model.dataset[cmtcols[0].name]
+    if not force:
```

Comment:
What should `force` do? I don't see any code for that case. Is it necessary? Perhaps `get_cmt` should create the column if it doesn't exist (not adding it to the dataset though).

Reply:
The idea behind force is to forcefully create a new CMT column (even if a previous one exists), as written in the docstring. All code needed for this is the `if not force:` statement. This functionality is used by the CMT update machinery to update previously dropped CMT columns which has since become outdated.
As `get_cmt()` extracts based on firstly the `compartment` datainfo type and secondly the name 'CMT', I thought this could be useful in this function as well. Since pharmpy will still save "CMT" columns in the dataset even if dropped, that column would be returned by `get_cmt()` (even if the user might not be interested in that one)